{railroad} Fails on SubExpressions

The switch statements in `railroad` don't handle `*ebnf.SubExpression`, so will panic if they are used.

This issue does not seem to be fixed.

`countProductions` still doesn't handle `*ebnf.SubExpression` in its switch.

Running with (7243a295a860f7c6d65ea9d7a74de53904d9524f):

```
panic: &ebnf.SubExpression{Expr: &ebnf.Expression{Alternatives: []*ebnf.Sequence{&ebnf.Sequence{Terms: []*ebnf.Term{&ebnf.Term{Token: "string", }}}, &ebnf.Sequence{Terms: []*ebnf.Term{&ebnf.Term{Token: "ident", }}}}}}

goroutine 1 [running]:
main.countProductions(0xc000187e78, 0x1195d88, 0xc000096c30, 0xb)
	participle/cmd/railroad/main.go:162 +0x5ea
main.countProductions(0xc000187e78, 0x1195da8, 0xc00019a2d0, 0x1)
	participle/cmd/railroad/main.go:157 +0x525
main.countProductions(0xc000187e78, 0x1195d68, 0xc0000a7110, 0x8)
	participle/cmd/railroad/main.go:150 +0x19b
main.countProductions(0xc000187e78, 0x1195d28, 0xc0000a7f98, 0x4)
	participle/cmd/railroad/main.go:146 +0xdb
main.countProductions(0xc000187e78, 0x1195d48, 0xc0000a7080, 0x0)
	participle/cmd/railroad/main.go:143 +0x457
main.countProductions(0xc000187e78, 0x1195d08, 0xc0000a67b0, 0x1195ea8)
	participle/cmd/railroad/main.go:135 +0x311
main.main()
	cmd/railroad/main.go:175 +0x219
```

--- a/cmd/railroad/main.go
+++ b/cmd/railroad/main.go
@@ -145,6 +145,8 @@ func countProductions(productions map[string]*production, n ebnf.Node) (size int
 		for _, a := range n.Alternatives {
 			size += countProductions(productions, a)
 		}
+	case *ebnf.SubExpression:
+		size += countProductions(productions, n.Expr)
 	case *ebnf.Sequence:
 		for _, t := range n.Terms {
 			size += countProductions(productions, t)

